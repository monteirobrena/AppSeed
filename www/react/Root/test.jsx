import React from "react";
import { configure, shallow, mount, render } from "enzyme";
import renderer from "react-test-renderer";
import "raf/polyfill";
import Adapter from "enzyme-adapter-react-16";
import Root from "./Root.jsx";

// Setup
// http://airbnb.io/enzyme/docs/api/
configure({ adapter: new Adapter() });

//--SNAP-SHOTS------
describe("Root (Snapshot)", () => {
  it("Root renders without crashing", () => {
    const component = renderer.create(<Root />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});

//--UI pieces------
describe("Root", () => {
  test("Shallow Rendering, make sure it renders something", () => {
    // Shallow rendering, component's children will NOT be rendered
    const wrapper = shallow(<Root />);
    expect(wrapper.length).toEqual(1);
  });

  test("Full Rendering", () => {
    // Full rendering: component's children WILL be rendered
    const wrapper = mount(<Root />);
    expect(wrapper.find(".application-container").exists()).toEqual(true);
  });
});

//--Business Logic------
