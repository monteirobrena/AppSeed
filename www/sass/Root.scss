#root {
    background-color: rgb(185, 185, 185);
    padding: 10px;
}