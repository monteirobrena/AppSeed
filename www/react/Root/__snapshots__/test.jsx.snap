// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Root (Snapshot) Root renders without crashing 1`] = `
<div
  className="application-container"
>
  <div>
    <ul
      className="react-component nav-bar sc-bdVaJa emdOWp"
    >
      <li
        className="sc-bwzfXH dHYqlz"
      >
        <a
          aria-current={false}
          className={undefined}
          href="/"
          onClick={[Function]}
          style={undefined}
        >
          View 1
        </a>
      </li>
      <li
        className="sc-bwzfXH dHYqlz"
      >
        <a
          aria-current={false}
          className={undefined}
          href="/view2"
          onClick={[Function]}
          style={undefined}
        >
          View 2
        </a>
      </li>
    </ul>
  </div>
  <div />
</div>
`;
